### PR TITLE
Combine apt-get in final docker image and remove cache

### PR DESCRIPTION
**Pull request by SuperSandro2000**
### Describe in detail the problem you had and how this PR fixes it

I had no problem but this reduces the amount of layers needed to download and removes the apt-get cache to reduce the image size

### Is there an open issue you can link to?

nop

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -22,9 +22,10 @@ COPY --from=0 /src/packages/server/cli-linux-x64 /usr/local/bin/code-server
 EXPOSE 8443
 RUN apt-get update && apt-get install -y \
 	openssl \
-	net-tools
-RUN apt-get install -y locales && \
-	locale-gen en_US.UTF-8
+	net-tools \
+	locales && \
+	locale-gen en_US.UTF-8 && \
+	rm -rf /var/lib/apt/lists/*
```
Comment (sr229) [inline Dockerfile]: you forgot to run `apt-get clean` first.

Comment (SuperSandro2000) [reply Dockerfile]: ``Official Debian and Ubuntu images automatically run apt-get clean, so explicit invocation is not required.``
Source: https://docs.docker.com/develop/develop-images/dockerfile_best-practices/

**Comment by nhooyr** [comment]
The two `RUN`s are intentionally separate for easier readability.

The apt-get cache removal looks reasonable to me.

**Comment by SuperSandro2000** [comment]
Combined it is readable too and there is already a comment. Don't think we need to keep it seperated.

If we don't combine it the cache removal is totally useless as we download all the data with the first layer and then remove it in the next layer. This means if we don't combine the apts we would make the image overall worse.

And personally I use Alpine for every image because it is so damn small and keeps my hard drive empty. I am currently doing them myself and maybe going to contribute them if they work good.

**Comment by nhooyr** [comment]
> Combined it is readable too and there is already a comment. Don't think we need to keep it separated.

Readability is subjective. I understand you think its readable but imo it is harder. Also, there is no comment.

> If we don't combine it the cache removal is totally useless as we download all the data with the first layer and then remove it in the next layer. This means if we don't combine the apts we would make the image overall worse.

That is true, but I think that's ok. The package cache is very small. I favour the readability of separate runs.

Thank you for the contribution but for the above reasons, I cannot merge it.

**Comment by SuperSandro2000** [comment]
Well then I do my own like always. People just can't make nice small docker images.

**Comment by coderasher** [comment]
@nhooyr Isn't it recommended to combine `update` and `install` commands? https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run

**Comment by ammario** [comment]
Yes, this PR is good at least because we want to couple the `apt-get update` layer with the `apt-get install` layer. It ensures the package cache is updated whenever we change which packages are installed.

**Comment by SuperSandro2000** [comment]
@coderasher oh thanks man! ❤️ I should get a printer and hang that on my wall.

**Comment by nhooyr** [comment]
> @nhooyr Isn't it recommended to combine update and install commands? https://docs.docker.com/develop/develop-images/dockerfile_best-practices/#run

Yes, but that would mean we add a `apt-get update` to the second `RUN`, not combine them.

> Yes, this PR is good at least because we want to couple the apt-get update layer with the apt-get install layer. It ensures the package cache is updated whenever we change which packages are installed.

We don't have to combine the RUNs to do that.

**Comment by SuperSandro2000** [comment]
> Yes, but that would mean we add a apt-get update to the second RUN, not combine them.

I am glad you didn't do that cause that would be kinda stupid.

**Comment by nhooyr** [comment]
@SuperSandro2000 Look man, we're all just trying to make code-server better. Can you please be a little more respectful?

I don't see why it would have been stupid. It would have been fine either way.

**Comment by SuperSandro2000** [comment]
English is not my first language so it is a bit hard for me. I didn't mean it in an offensive way or so.

We would basically run the same command twice. I don't understand why you don't like the idea of just merging the two lines. Readability is just fine.

**Comment by nhooyr** [comment]
> English is not my first language so it is a bit hard for me. I didn't mean it in an offensive way or so.

Understood. It's all good.

**Pull request closed without merging**